### PR TITLE
Add mint trait

### PR DESCRIPTION
**Pull request by Henry-E**
Going to try to add copy the code used for creating TokenAccount in the account constraints and use it to also initialize Mints.

This is a new branch but it still contains some of the local changes I need to run anchor. I'll have to remove those commits before this can be pulled, if/once it's done.

### CODE REVIEW

File: cli/Cargo.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ serde = { version = "1.0.122", features = ["derive"] }
 solana-sdk = "1.7.4"
 solana-program = "1.7.4"
 solana-client = "1.7.4"
-serum-common = { git = "https://github.com/project-serum/serum-dex", features = ["client"] }
+serum-common = { path = "../../prediction_market/deps/serum-dex", features = ["client"] }
```
Comment (tomland123) [inline cli/Cargo.toml:29]: Don't think you can change this line lol

File: examples/compute-cost/program_logs/EvQ2hf9K3C6UuoMoseaGdrAiFxqd6AGYBacXrfwB8qv5.compute_cost.log [hunk 2]
```diff
@@ -0,0 +1,242 @@
+Streaming transaction logs mentioning EvQ2hf9K3C6UuoMoseaGdrAiFxqd6AGYBacXrfwB8qv5. Confirmed commitment
```
Comment (tomland123) [inline examples/compute-cost/program_logs/EvQ2hf9K3C6UuoMoseaGdrAiFxqd6AGYBacXrfwB8qv5.compute_cost.log:1]: are these logs not gitignored? @armaniferrante

Comment (Henry-E) [reply examples/compute-cost/program_logs/EvQ2hf9K3C6UuoMoseaGdrAiFxqd6AGYBacXrfwB8qv5.compute_cost.log:1]: I'm going to remove all the commits unrelated to the mint changes

**Comment by Henry-E** [comment]
Going to make a new pull request with a cleaner branch

**Pull request closed without merging**